Shared library for eclipse builds.

seiferma opened the pull request:
This is the initial implementation of the shared library to be used by the mdsd.tools builds. It covers
* checkout
* build
* quality checks
* deployment
* notifications

The steps are executed on slaves instead of the master. This allows us to use cheap hosts as build slaves and a more reliable host as master.

Builds are done inside of a docker container using Java 11. The container has limitations in time, usable hdd, and ram. This is important to prevent malicious pull requests from breaking the slave. All builds that are not pull requests can write to a shared maven cache, while pull requests can only read the cache.

As part of quality metrics, we collect Javadoc, checkstyle issues, junit results, and jacoco coverage.

The deployment takes place via SSH. If a release is scheduled, the nightly build will be copied to a named release folder and a shell script will be executed that generated a composite update site for all releases.

If a build fails, the last committer and a default recipient will be notified. If the build continues to fail, all commiters since the initial broken commit will be notified.

A demonstration of the running script is available on the [build host](https://build.mdsd.tools/job/MDSD-Tools/job/CDODebugUtils/job/jenkinsfile/). We should clean up the [deployment target](https://updatesite.mdsd.tools/) after merging of this PR.

@sdkrach You are familar with fancy programming languages. You might have better ideas on how to structure and clean up the code.

If we merge this PR, I prefer a squash merge because creating this pipeline required some heavy use of try and error. Single commits are mostly nonsense.

seiferma committed:
First try

seiferma committed:
try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Tries.

seiferma committed:
try.

sdkrach reviewed (changes requested):
Nice work! At some point in the future we might improve the extensibility of the build process. Currently, I can't think of a use case that you wouldn't use Maven for, anyhow. 

One remark: I'd prefer being able to set the email address to receive the notifications in the Jenkins file. Projects in a very early stage might not want the entire mdsd.tools user community to be notified of every project instability.

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Try

seiferma committed:
Update mdsdToolsEclipsePipeline.groovy

seiferma commented:
I added a configuration parameter that can take a plain email or a base64 encoded email. In addition, I changed the default recipient to the mdsd-tools build mailing list.

sdkrach reviewed (approved):
Looks good to me now
